sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernatePoolOrderCompletionDao.java

```diff
@@ -0,0 +1,67 @@
+package uk.ac.bbsrc.tgac.miso.persistence.impl;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+
+import org.hibernate.HibernateException;
+import org.hibernate.Query;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import uk.ac.bbsrc.tgac.miso.core.data.PoolOrderCompletion;
+import uk.ac.bbsrc.tgac.miso.core.store.PlatformStore;
+import uk.ac.bbsrc.tgac.miso.core.store.PoolStore;
+import uk.ac.bbsrc.tgac.miso.persistence.PoolOrderCompletionDao;
+
+@Transactional
+@Repository
+public class HibernatePoolOrderCompletionDao implements PoolOrderCompletionDao {
+
+  @Autowired
+  private SessionFactory sessionFactory;
+
+  @Autowired
+  @Qualifier("sqlPlatformDAO")
```

Comment:
If you rename these fields to platformStore and poolStore, it maps them correctly without the need for the qualifier annotation... Which I think it better. I'm still not sure what purpose the ProxyFactoryInterfaces serve, and they are what causes this problem